polyfill/index.d.ts

```diff
@@ -0,0 +1,385 @@
+    microseconds?: number;
+    nanoseconds?: number;
+  };
+
+  export class Duration extends Required<DurationLike> {
+    static from(
+      item: Temporal.Duration | DurationLike | string | object,
+      options?: AssignmentOptions
+    ): Temporal.Duration;
+    constructor(
+      years?: number,
+      months?: number,
+      days?: number,
+      hours?: number,
+      minutes?: number,
+      seconds?: number,
+      milliseconds?: number,
+      microseconds?: number,
+      nanoseconds?: number
+    );
+    readonly years: number;
+    readonly months: number;
+    readonly days: number;
+    readonly hours: number;
+    readonly minutes: number;
+    readonly seconds: number;
+    readonly milliseconds: number;
+    readonly microseconds: number;
+    readonly nanoseconds: number;
+    with(durationLike: DurationLike, options: AssignmentOptions): Temporal.Duration;
+    plus(other: Temporal.Duration, options: ArithmeticOptions): Temporal.Duration;
+    minus(other: Temporal.Duration, options: DurationMinusOptions): Temporal.Duration;
+    getFields(): Required<DurationLike>;
+    toLocaleString(locales?: string | string[], options?: Intl.DateTimeFormatOptions): string;
+    toJSON(): string;
+    toString(): string;
+  }
+
+  export class Absolute extends Required<AbsoluteLike> {
+    static fromEpochSeconds(epochSeconds: number): Temporal.Absolute;
+    static fromEpochMilliseconds(epochMilliseconds: number): Temporal.Absolute;
+    static fromEpochMicroseconds(epochMicroseconds: bigint): Temporal.Absolute;
+    static fromEpochNanoseconds(epochNanoseconds: bigint): Temporal.Absolute;
+    static from(item: Temporal.Absolute | string | object): Temporal.Absolute;
+    static compare(one: Temporal.Absolute, two: Temporal.Absolute): 1 | -1 | 0;
+    constructor(epochNanoseconds: bigint);
+    getEpochSeconds(): number;
+    getEpochMilliseconds(): number;
+    getEpochMicroseconds(): bigint;
+    getEpochNanoseconds(): bigint;
+    plus(temporalDurationLike: DurationLike): Temporal.Absolute;
+    minus(temporalDurationLike: DurationLike): Temporal.Absolute;
+    difference(
+      other: Temporal.Absolute,
+      options?: DifferenceOptions<'days' | 'hours' | 'minutes' | 'seconds'>
+    ): Temporal.Duration;
+    // Should Absolute and DateTime have different names for `inTimeZone`? See #574.
+    inTimeZone(temporalTimeZoneLike?: TimeZoneLike): Temporal.DateTime;
+    toLocaleString(locales?: string | string[], options?: Intl.DateTimeFormatOptions): string;
+    toJSON(): string;
+    toString(temporalTimeZoneLike?: TimeZoneLike): string;
+  }
+
+  export type DateLike = {
+    year?: number;
+    month?: number;
+    day?: number;
+  };
+
+  export class Date extends Required<DateLike> {
+    static from(item: Temporal.Date | string | object, options?: AssignmentOptions): Temporal.Date;
+    static compare(one: Temporal.Date, two: Temporal.Date): ComparisonResult;
+    constructor(year: number, month: number, day: number);
```

Comment:
We are calling these constructor arguments `isoYear`, `isoMonth`, and `isoDay` in the spec to emphasize that they are not calendar-aware. They haven't been renamed in the polyfill yet, but probably best to put the correct names in the types file to start with.

Reply:
One concern I had for that naming was that "ISO" for many JS devs means "_that weird string format that ends in Z and has the wrong time that I don't really understand but I know it's used to store in databases or send to external APIs_". So just using "iso" alone will probably confuse more than it will help. My suggestion would be to leave as "year" in the parameter name and then to highlight in the docs that it's an ISO Calendar year. Per @littledan's [notes](https://github.com/tc39/proposal-temporal/issues/268#issuecomment-570670212) about who's actually going to be developing with non-ISO calendars, it seems clear that (unlike, for example, Japanese language localization where the developer may only know Japanese) most developers using non-ISO calendars are also using ISO calendars, so they'll probably already be on the lookout for docs like this because they're already familiar with the challenges of dual-calendar implementations. 

What do you think?

If you do want to change the parameter names, then I'd at least suggest `isoCalendarXXX` to clarify that ISO in this context refers to the ISO calendar and not the ISO string format.

Reply:
@ptomato - I wanted to circle back after our discussion in the other issue. First, should these parameters be renamed in the TS types?  If yes: 
* should they be 'isoXxx' or 'isoCalendarXxx'?  I think the latter might be a little clearer but don't feel very strongly one way or the other. 
* which types' constructor YMD parameters should be renamed? All of them or just a subset?

Reply:
I think `isoCalendarXxx` is not necessary for just the constructor arguments, since we are considering the constructors to be low-level API. And all of them should be renamed, since those types will all support other calendars. So my suggestion would be:
- Date - `constructor(isoYear: number, isoMonth: number, isoDay: number)`
- DateTime - `constructor(isoYear: number, isoMonth: number, isoDay: number, hour?: number, minute?: number, second?: number, millisecond?: number, microsecond?: number, nanosecond?: number)`
- YearMonth - `constructor(isoYear: number, isoMonth: number)`
- MonthDay - `constructor(isoMonth: number, isoDay: number)`

Reply:
OK sound good. Making changes now. I'm assuming that `YearMonth.prototype.withDay` and `MonthDay.prototype.withYear` can stay as plain `day` and `year`, right?

Reply:
That's right, if the calendar is not ISO, then they take calendar days/years, not ISO days/years.